sources/commands/option_external.c

```diff
@@ -6,20 +6,25 @@
 /*   By: inikulin <inikulin@student.42berlin.de>    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/09/30 00:10:22 by inikulin          #+#    #+#             */
-/*   Updated: 2024/10/12 01:19:09 by inikulin         ###   ########.fr       */
+/*   Updated: 2024/10/21 03:14:32 by inikulin         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "commands.h"
 
-static char	*find_executable(char *tgt, t_dlist *path)
+static char	*find_executable(char *tgt, t_dlist *path, int *errno)
 {
 	char	*fullpath;
 
 	while (path)
 	{
 		fullpath = ft_strjoin_multi_free_outer(
 				ft_s2(path->content, tgt), 2, "/");
+		if (!fullpath)
+		{
+			ft_assign_i(errno, 4, 0);
+			return (0);
+		}
 		if (access(fullpath, X_OK) == 0)
```

Comment:
3.2 if such a file exists and is available for execution, returns its full name (to execute it outside)